main.js

```diff
@@ -52,10 +52,40 @@ function RoundToHundredths(x) {
   return Math.floor(x * 100 + 0.5) / 100;
 }
 
-function HandleNode(svgNode, scaleX, scaleY, translateX, translateY) {
+// |fillString| is expected to be "#RRGGBB" or "#RGB"].
+function PathColorFromFill(fillString) {
+  if (fillString.length === 4) {
+    // Color in form of #RGB so let's turn that to #RRGGBB.
+    fillString = `#${fillString[1]}${fillString[1]}${fillString[2]}${fillString[2]}${fillString[3]}${fillString[3]}`.toUpperCase();
+  }
+
+  const r = fillString.substr(1,2);
+  const g = fillString.substr(3,2);
+  const b = fillString.substr(5,2);
+  
+  return `PATH_COLOR_ARGB, 0xFF, 0x${r}, 0x${g}, 0x${b},\n`;
+}
+
+function HandleNode(svgNode, scaleX, scaleY, translateX, translateY, preserveFill) {
   var output = '';
   for (var idx = 0; idx < svgNode.children.length; ++idx) {
     var svgElement = svgNode.children[idx];
+
+    if (preserveFill) {
```

Comment:
Hmmm, do we really want this block here before we know that we're dealing with a path? Wouldn't that mean we're adding NEW_PATH for some other element like <g>

Reply:
This is a great point. I agree and will update so that a NEW_PATH isn't issued for the \<g>. I think that makes tons of sense, considering a child element inside <g> should be the one needing the NEW_PATH, not the \<g> itself

Something funny I found along the way, [according to MDN](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/g), the following is valid SVG:
```
<svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
  <!-- Using g to inherit presentation attributes -->
  <g fill="white" stroke="green" stroke-width="5">
    <circle cx="40" cy="40" r="25" />
    <circle cx="60" cy="60" r="25" />
  </g>
</svg>
```

The fill, stroke, and width of the \<g> element are inherited by the children elements. So it seems this data would be lost if converted with the tool currently.

Reply:
True. There is a LOT of SVG that isn't parsed by this script though. We're nowhere near SVG-complete.

Reply:
Have updated, this path won't run for <g>, will run for rect / circles. That makes sense to me but lmk what you think